text/0000-types-team.md

```diff
@@ -0,0 +1,240 @@
+- Feature Name: N/A
+- Start Date: 2022-04-12
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+* Introduce a new "type system team" (the "types team" for short) that works to **implement and formally define** the semantics of Rust as decided by the lang team.
+* The type team owns and maintains:
+    * The implementation of the Rust type checker, trait system, borrow checker that is used in rustc.
+    * Formal definitions of Rust, its type checker, and its semantics, as they are developed.
+    * The "unsafe code guidelines" (once decided).
+
+# Motivation
+[motivation]: #motivation
+
+The types team is meant to build a base of maintainers for the formal side of Rust, both design and implementation. This has traditionally been an area with a very low "[bus factor]", both in terms of the compiler (few maintainers to the code) and the language design (few people who fully understand the entire space). Focusing a team on just Rust's "type system" will allow us to do targeted outreach and to help people to learn the background that is needed to contribute here.
+
+Like the compiler team, the team's plays a "supportive" role. Its goal is to ensure both that Rust's type system has an efficient and correct implementation and that the type system (design and implementation) is able to scale to new demands and requirements. At this moment, Rust is suffering from a general paralysis in which new features (implied bounds, const generics, specialization, etc) are stalled for long periods of time due to a combination of an inflexible implementation, a lack of maintainers, and a general difficulty in reasoning about their interactions.
+
+[bus factor]: https://en.wikipedia.org/wiki/Bus_factor
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Mission and charter
+
+The “types team” owns the design, implementation, and maintenance of the Rust type system (but not the user-facing syntax of Rust). This includes the Rust type checker, trait system, borrow checker, and the operational semantics of MIR (and eventually Rust itself). As part of the operational semantics, the team is also ultimately responsible for deciding what is “undefined behavior”.
+
+## Relationship to other teams
+
+The types team is conceptually a subteam of both the lang and compiler teams. Since the team repo can't really model that, we'll put it primarily under the lang team.
+
+### Lang team
+
+* Stabilizing new features that intersect the type system will require "types team approval".
+    * Eventually this is expected to require extending the "formality" models to include the feature.
+    * This approval is not meant to be used to make "policy" decisions but to enforce the soundness and implementability of the feature itself.
+* Advising on how to soundness bugs or other subtle questions arise.
```

Comment:
"Advising on how to _address_ soundness bugs or other subtle questions _that_ arise" perhaps?

Reply:
```suggestion
* Advising on how to address soundness bugs or other subtle questions that arise.
```